src/main/kotlin/no/nav/familie/ef/sak/service/VurderingService.kt

```diff
@@ -110,83 +155,112 @@ class VurderingService(private val behandlingService: BehandlingService,
                                                      metadata: HovedregelMetadata): List<Vilkårsvurdering> {
         val lagredeVilkårsvurderinger = vilkårsvurderingRepository.findByBehandlingId(behandlingId)
 
-        if (behandlingErLåstForVidereRedigering(behandlingId)) {
-            return lagredeVilkårsvurderinger
+        return when {
+            behandlingErLåstForVidereRedigering(behandlingId) -> lagredeVilkårsvurderinger
+            lagredeVilkårsvurderinger.isEmpty() -> lagreNyeVilkårsvurderinger(behandlingId, metadata)
+            else -> lagredeVilkårsvurderinger
         }
+    }
 
-        if (lagredeVilkårsvurderinger.isEmpty()) {
-            return opprettNyeVilkårsvurderinger(behandlingId, metadata)
-        } else {
-            return lagredeVilkårsvurderinger
-        }
+    private fun lagreNyeVilkårsvurderinger(behandlingId: UUID,
+                                           metadata: HovedregelMetadata): List<Vilkårsvurdering> {
+        val nyeVilkårsvurderinger: List<Vilkårsvurdering> = opprettNyeVilkårsvurderinger(behandlingId, metadata)
+        return vilkårsvurderingRepository.insertAll(nyeVilkårsvurderinger)
     }
 
-    private fun opprettNyeVilkårsvurderinger(behandlingId: UUID,
-                                             metadata: HovedregelMetadata): List<Vilkårsvurdering> {
-        val søknad = metadata.søknad
-        val nyeVilkårsvurderinger: List<Vilkårsvurdering> = alleVilkårsregler
+    fun opprettNyeVilkårsvurderinger(behandlingId: UUID,
```

Comment:
Skal denne være private?

Reply:
Denne ble endret til publik for å kunne bli brukt fra tester. Egentlige kan den plukkes ut til en util-klasse eller noe i stedet då det kan være static